core/services/pipeline/runner.go

```diff
@@ -274,14 +276,24 @@ func (r *runner) run(
 	}
 
 	for taskRun := range scheduler.taskCh {
+		taskRun := taskRun
 		// execute
-		go func(taskRun *memoryTaskRun) {
+		go recovery.WrapRecoverHandle(l, func() {
 			result := r.executeTaskRun(ctx, run.PipelineSpec, taskRun, l)
 
 			logTaskRunToPrometheus(result, run.PipelineSpec)
 
 			scheduler.report(reportCtx, result)
-		}(taskRun)
+		}, func(err interface{}) {
+			t := time.Now()
+			scheduler.report(reportCtx, TaskRunResult{
+				ID:         uuid.NewV4(),
+				Task:       taskRun.task,
+				Result:     Result{Error: ErrRunPanicked{err}},
+				FinishedAt: null.TimeFrom(t),
+				CreatedAt:  t, // TODO: more accurate start time
+			})
+		})
```

Comment:
@archseer Does this effectively reapply the recovery removed here?
https://github.com/smartcontractkit/chainlink/pull/5384/files?diff=unified&w=0#diff-e219e9df67a3181e6e8f0b4806f90b582df4b8d4dc8d3ec85ea132f5cf2dfe9fL273-L286